File: Lab6Calificado/Lab06_2026_1_Plantilla/Assessment/Persistance/src/main/java/pe/edu/pucp/assessment/dao/persistance/IDAO.java
package pe.edu.pucp.assessment.dao.persistance;

import java.util.List;

public interface IDAO <T> {
    int insertar(T objeto);
    int modificar(T objeto);
    int eliminar(int id);
    T buscarPorId(int id);
    List<T> listarTodos();
}

File: Lab7Calificado/Lab7_2026_1/PokeSoftSolution/PokeSoftPersistance/DAO/IDAO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PokeSoftPersistance.DAO
{
    public interface IDAO <T>
    {
        int Insertar(T objeto);
        int Modificar(T objeto);
        int Eliminar(int id);
        T BuscarPorId(int id);
        List<T> ListarTodos();
    }
}
